### DISCUSSION TITLE
Is there documentation for the import file formats?

### DISCUSSION THREAD

**Discussion by rmohns** [ArchiveBox/ArchiveBox#1182] [Q&A]
The Wiki says that JSON, XML, HTML, and text file imports are supported, but does not describe what format they should be in. While one-URL-per-line txt is obvious enough, by their nature JSON, XML and HTML could be _anything_.

I've searched the wiki and discussions and all I've found is [this](https://docs.archivebox.io/en/master/Usage.html#import-a-list-of-urls-from-a-text-file):

> You can also pipe in RSS, XML, Netscape, or any of the other supported import formats via stdin.
> 
> ``` 
> archivebox add < ~/Downloads/browser_bookmarks_export.html
> # or
> archivebox add < ~/Downloads/pinboard_bookmarks.json
> # or
> archivebox add < ~/Downloads/other_links.txt
> ``` 

But I haven't found any documentation on _how those files should be structured_. I'm having real trouble getting ArchiveBox to import my Pinboard bookmarks with tags intact. I've tried Delicious-1.0 format XML, JSON, and Netscape HTML Bookmark file (ick). None work.

I'm willing to transform what I have to a format ArchiveBox likes, but I don't know what those formats should look like.

**Comment by hugochisholm** [comment]
I was having a similar issue recently so figured I'd chime in. There is a solution that's been implemented currently in the `:dev` release. See issue #725 and PR #911. I can confirm that I've had success with Pinboard JSON format tag importing using that method.

That said, I'd be interested in being able to import the bookmarked date _from_ the Pinboard JSON to be used as the bookmarked date in Archivebox. I've had that account since 2010 and, for me at least, retaining those dates would be of value to me.

**Reply by pirate** [reply]
Retaining the import dates is definitely desired, unfortunately so many sources export different date formats that's it's really hard to do one size fits all. 

This has been a long running struggle with the parsers, even supposedly standard formats sometimes import with offsets from some sources and trying to un-offset them results in other sources getting messed up. 

I'm happy to review any PRs to improve the existing import logic, any help here is very welcome. 🙏😊
